### ISSUE TITLE
Compilation Error

### ISSUE THREAD

**Issue by lbhdc** [lobachevzky/mujoco#5]
I am having some trouble installing this package.

My general system info:
```
$uname -ori
4.13.0-36-generic x86_64 GNU/Linux
```

```
$gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
$python --version
Python 2.7.12
$python3 --version
Python 3.5.2
```
Let me know if you need additional system info.

Here are the steps I have taken so far:
```
$cd /git/clone/dir
$make
python setup.py build_ext --inplace
Traceback (most recent call last):
  File "setup.py", line 5, in <module>
    from Cython.Build import cythonize
ImportError: No module named Cython.Build
Makefile:7: recipe for target 'default' failed
make: *** [default] Error 1

```
I changed two lines in the `Makefile`
line `7` to `python3 setup.py build_ext --inplace`
line `33` to `python3 setup.py bdist_wheel`

After I saved those changes I ran `make` again and it gave the following:
```
python3 setup.py build_ext --inplace
---------------------------
Using the following config:
path to mjkey.txt: ~/.mujoco/mjkey.txt
mjpro150 directory: ~/.mujoco/mjpro150
directory containing libOpenGL.so(should be None if you don't have a GPU): None
To change, edit /home/<user_name>/programs/mj_py_wrapper/config.yml
---------------------------
Compiling mujoco/glfw.pyx because it changed.
[1/1] Cythonizing mujoco/glfw.pyx
running build_ext
building 'mujoco.glfw' extension
creating build/temp.linux-x86_64-3.5
creating build/temp.linux-x86_64-3.5/build
creating build/temp.linux-x86_64-3.5/build/mujoco
creating build/temp.linux-x86_64-3.5/src
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DMJKEY_PATH="/home/<user_name>/.mujoco/mjkey.txt" -I/home/<user_name>/.mujoco/mjpro150/include -I/usr/local/lib/python3.5/dist-packages/numpy/core/include -Iheaders -Ipxd -I/usr/include/python3.5m -c build/mujoco/glfw.c -o build/temp.linux-x86_64-3.5/build/mujoco/glfw.o -Wno-unused-function -std=c99
In file included from /usr/local/lib/python3.5/dist-packages/numpy/core/include/numpy/ndarraytypes.h:1809:0,
                 from /usr/local/lib/python3.5/dist-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                 from /usr/local/lib/python3.5/dist-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from build/mujoco/glfw.c:586:
/usr/local/lib/python3.5/dist-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DMJKEY_PATH="/home/<user_name>/.mujoco/mjkey.txt" -I/home/<user_name>/.mujoco/mjpro150/include -I/usr/local/lib/python3.5/dist-packages/numpy/core/include -Iheaders -Ipxd -I/usr/include/python3.5m -c src/utilGlfw.c -o build/temp.linux-x86_64-3.5/src/utilGlfw.o -Wno-unused-function -std=c99
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DMJKEY_PATH="/home/<user_name>/.mujoco/mjkey.txt" -I/home/<user_name>/.mujoco/mjpro150/include -I/usr/local/lib/python3.5/dist-packages/numpy/core/include -Iheaders -Ipxd -I/usr/include/python3.5m -c src/util.c -o build/temp.linux-x86_64-3.5/src/util.o -Wno-unused-function -std=c99
creating build/lib.linux-x86_64-3.5
creating build/lib.linux-x86_64-3.5/mujoco
x86_64-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-Bsymbolic-functions -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.5/build/mujoco/glfw.o build/temp.linux-x86_64-3.5/src/utilGlfw.o build/temp.linux-x86_64-3.5/src/util.o -L/home/<user_name>/.mujoco/mjpro150/bin -lmujoco150 -lGL -lglew -o build/lib.linux-x86_64-3.5/mujoco/glfw.cpython-35m-x86_64-linux-gnu.so -fopenmp /home/<user_name>/.mujoco/mjpro150/bin/libglfw.so.3
copying build/lib.linux-x86_64-3.5/mujoco/glfw.cpython-35m-x86_64-linux-gnu.so -> mujoco
```

I ran this a few times. I purged `/usr/local/lib/python3.5/dist-packages/mujoco*` and `make clean` however it still returns the following:
```
$pip3 install -e .
Requirement already satisfied: Cython>=0.27.3 in /usr/local/lib/python3.5/dist-packages (from mujoco==2.0.2)
Requirement already satisfied: numpy==1.13.3 in /usr/local/lib/python3.5/dist-packages/numpy-1.13.3-py3.5-linux-x86_64.egg (from mujoco==2.0.2)
Requirement already satisfied: pyyaml>=3.12 in /usr/local/lib/python3.5/dist-packages (from mujoco==2.0.2)
Installing collected packages: mujoco
  Found existing installation: mujoco 2.0.2
    Can't uninstall 'mujoco'. No files were found to uninstall.
  Running setup.py develop for mujoco
Successfully installed mujoco
```
I ran this `test_script.py` to verify my install:
```
import os
os.environ["EGL"] = '1'
import mujoco
```
iPython returns:
```
runfile('test_script.py', wdir='/home/<user_name>/programs/projects/', post_mortem=True)
Using EGL version of mujoco.

Error compiling Cython file:
------------------------------------------------------------
...
include "mujoco/sim.pyx"
^
------------------------------------------------------------

/usr/local/lib/python3.5/dist-packages/mujoco/egl.pyx:1:0: 'mujoco/sim.pyx' not found

Traceback (most recent call last):

  File "<ipython-input-20-12872c74c731>", line 1, in <module>
    runfile('test_script.py', wdir='/home/<user_name>/programs/projects', post_mortem=True)

  File "/usr/local/lib/python3.5/dist-packages/spyder/utils/site/sitecustomize.py", line 705, in runfile
    execfile(filename, namespace)

  File "/usr/local/lib/python3.5/dist-packages/spyder/utils/site/sitecustomize.py", line 102, in execfile
    exec(compile(f.read(), filename, 'exec'), namespace)

  File "test_script.py", line 13, in <module>
    import mujoco

  File "/usr/local/lib/python3.5/dist-packages/mujoco/__init__.py", line 18, in <module>
    from mujoco.egl import SimEgl as Sim, GeomType, ObjType

  File "/usr/local/lib/python3.5/dist-packages/pyximport/pyximport.py", line 458, in load_module
    language_level=self.language_level)

  File "/usr/local/lib/python3.5/dist-packages/pyximport/pyximport.py", line 231, in load_module
    raise exc.with_traceback(tb)

  File "/usr/local/lib/python3.5/dist-packages/pyximport/pyximport.py", line 215, in load_module
    inplace=build_inplace, language_level=language_level)

  File "/usr/local/lib/python3.5/dist-packages/pyximport/pyximport.py", line 191, in build_module
    reload_support=pyxargs.reload_support)

  File "/usr/local/lib/python3.5/dist-packages/pyximport/pyxbuild.py", line 102, in pyx_to_dll
    dist.run_commands()

  File "/usr/lib/python3.5/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)

  File "/usr/lib/python3.5/distutils/dist.py", line 974, in run_command
    cmd_obj.run()

  File "/usr/local/lib/python3.5/dist-packages/Cython/Distutils/old_build_ext.py", line 186, in run
    _build_ext.build_ext.run(self)

  File "/usr/lib/python3.5/distutils/command/build_ext.py", line 338, in run
    self.build_extensions()

  File "/usr/local/lib/python3.5/dist-packages/Cython/Distutils/old_build_ext.py", line 194, in build_extensions
    self.build_extension(ext)

  File "/usr/lib/python3.5/distutils/command/build_ext.py", line 532, in build_extension
    depends=ext.depends)

  File "/usr/lib/python3.5/distutils/ccompiler.py", line 574, in compile
    self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)

  File "/usr/lib/python3.5/distutils/unixccompiler.py", line 120, in _compile
    raise CompileError(msg)

ImportError: Building module mujoco.egl failed: ["distutils.errors.CompileError: command 'x86_64-linux-gnu-gcc' failed with exit status 1\n"]

****************************************
Entering post mortem debugging...
****************************************
> /usr/lib/python3.5/distutils/unixccompiler.py(120)_compile()
    118                        extra_postargs)
    119         except DistutilsExecError as msg:
--> 120             raise CompileError(msg)
    121 
    122     def create_static_lib(self, objects, output_libname,
```
Any help would be appreciated.

**Comment by lobachevzky** [comment]
Ok. I will update the README to address this more explicitly.

The order of operations is as follows:
1. `pip install cython`. Without this, the build will fail.
2. `make`
3. `pip install .`

The problem you are encountering is that when you first run `make`, you don't have Cython so `make` fails. Then since `make` failed to compile the Cython file, `pip install` also fails.

Let me know if this helps.

**Comment by lbhdc** [comment]
That was weird. Pip was reporting that I had it installed, but after I reinstalled it it seemed to clear up some of the issues I was having. 

I had a few other problems, but I believe those were due to my mistakes while troubleshooting.
I had inadvertently altered the `$LD_LIBRARY_PATH` so that it wasn't correctly pointing to my 'mjpro150/bin'. After correcting that, I also had an error `config.yml`. I am using nvidia drivers, and was trying to use `/usr/include/GL` instead of /usr/lib/nvidia-384`.

I am now able to run in EGL or GLFW versions with no errors. Thanks for your help!
